### PR TITLE
Bunch of patches to better support IOSXE platform

### PR DESCRIPTION
**Pull request by ddutt**
Based on data from real world by a user.

### CODE REVIEW

File: suzieq/poller/worker/services/interfaces.py [hunk 1]
```diff
@@ -824,21 +836,30 @@ def _clean_iosxr_data(self, processed_data, raw_data):
...
+            entry['type'] = iftype
             # This is specifically for IOSXE/IOS devices where
             # the IPv6 address uses capital letters
             if devtype != 'iosxr':
                 entry['ip6AddressList'] = [x.lower()
                                            for x in entry.get('ip6AddressList',
                                                               [])]
 
+            if entry['ifname'].endswith('.0'):
+                entry['vlan'] = -1
```
Comment (LucaNicosia) [inline suzieq/poller/worker/services/interfaces.py:855]: Use a constant defined inside utils instead of '-1'?

Comment (ddutt) [reply suzieq/poller/worker/services/interfaces.py:855]: Ack, I agree

File: suzieq/engines/pandas/interfaces.py [hunk 2]
```diff
@@ -270,7 +271,7 @@ def _check_ipaddr(x, fld1, fld2, reason):
                     .fillna({i: [] for i in if_df.index})
 
         if 'vlanList' not in if_df.columns:
-            if_df['vlanList'] = []
+            if_df['vlanList'] = [[] for i in repeat(None, len(if_df))]
```
Comment (claudiolor) [inline suzieq/engines/pandas/interfaces.py:274]:
```suggestion
            if_df['vlanList'] = list(repeat([], len(if_df)))
```
If we use itertools something like this should be enough.
Or without using any extension:
```suggestion
            if_df['vlanList'] = [[] for i in range(len(if_df))]
```

File: suzieq/poller/worker/coalescer_launcher.py [hunk 3]
```diff
@@ -135,7 +137,7 @@ async def _start_coalescer(self) -> Process:
         else:
             coalescer_args_str = ''
         coalescer_args_str = f'{self.coalescer_bin} {coalescer_args_str}'
-        coalescer_args = coalescer_args_str.strip().split()
+        coalescer_args = shlex.split(coalescer_args_str.strip())
```
Comment (claudiolor) [inline suzieq/poller/worker/coalescer_launcher.py:140]: If the problem is building the list from the string, wouldn't be better to directly create a list? So having something like this:
```python
coalescer_args = [self.coalescer_bin]
if self.config_file:
   coalescer_args += ['-c', self.config_file]
```

File: suzieq/poller/worker/services/vlan.py [hunk 4]
```diff
@@ -1,4 +1,5 @@
 import numpy as np
+from collections import defaultdict
```
Comment (claudious96) [inline suzieq/poller/worker/services/vlan.py:2]: This import seems to be unused

File: suzieq/poller/worker/services/macs.py [hunk 5]
```diff
@@ -150,10 +150,18 @@ def _clean_nxos_data(self, processed_data, _):
 
     def _clean_iosxe_data(self, processed_data, _):
 
+        foo = ''
```
Comment (claudious96) [inline suzieq/poller/worker/services/macs.py:153]: Leftover here ?

**Review by claudious96 — APPROVED** [review]

File: suzieq/engines/pandas/interfaces.py (hunk 2 above)
Comment (ddutt) [reply suzieq/engines/pandas/interfaces.py:274]: Ack

File: suzieq/poller/worker/coalescer_launcher.py (hunk 3 above)
Comment (ddutt) [reply suzieq/poller/worker/coalescer_launcher.py:140]: Sure, that'd work too. This seems simpler. The user in this case had a dir with weird characters in it such as (, \, space etc.

File: suzieq/poller/worker/services/vlan.py (hunk 4 above)
Comment (ddutt) [reply suzieq/poller/worker/services/vlan.py:2]: Yes, it was removed

**Pull request merged**